src/modules/c.rs

```diff
@@ -0,0 +1,220 @@
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::c::CConfig;
+use crate::formatter::StringFormatter;
+use crate::formatter::VersionFormatter;
+
+use once_cell::sync::Lazy;
+use std::borrow::Cow;
+use std::ops::Deref;
+
+/// Creates a module with the current c version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("c");
+    let config: CConfig = CConfig::try_load(module.config);
+    let is_c_project = context
+        .try_begin_scan()?
+        .set_extensions(&config.detect_extensions)
+        .set_files(&config.detect_files)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_c_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        let c_compiler_info = Lazy::new(|| {
+            context.exec_cmds_return_first(&[
+                // the compiler is usually cc, and --version works on gcc and clang
+                &["cc", "--version"],
+                // but on some platforms gcc is installed as *gcc*, not cc
+                &["gcc", "--version"],
+                // for completeness, although I've never seen a clang that wasn't cc
+                &["clang", "--version"],
```

Comment:
I think it might be nice to make this configurable. Some people might use compiler variants like `zig cc` and `arm-none-eabi-gcc`.

Reply:
It would be nice, but I'm not sure how practical it is to make it entirely configurable. We'd need to have a way of specifying what command to run, with what arguments, what to look for in the output, and possibly how to parse the version out of that.